pcsx2/gui/FrameForGS.cpp

```diff
@@ -438,7 +438,7 @@ GSFrame::~GSFrame() throw()
 void GSFrame::OnCloseWindow(wxCloseEvent& evt)
 {
 	sApp.OnGsFrameClosed( GetId() );
-	Hide();		// and don't close it.
+	CoreThread.Reset(); // reset thread to avoid suspend status on close event.
```

Comment:
> Suspending alone will not fix the memory card dialog issue since suspend still means CoreThread.IsRunning() will return true

Assuming you're right, the solution would be to find a function which will tell you whether or not it's suspended.

> Reset is the only solution I've managed to find. Reset can be considered as the same function as "shutdown" in system drop down list so I don't believe it could cause any issues but it's probable though.

Not sure I understand this. You say that `CoreThread.Reset()` is roughly the same as shutdown, and you don't care that the user never wanted to shutdown emulation?

There should be a single goal here: if we were suspended when opening the MCD dialog, then don't resume once the MCD dialog is closed.

We most probably don't need to change what happens when the GS window is closed, nor change what happens when the MCD dialog is closed if the emulation was running while it was opened. We only need to not resume if emulation was not running when it was opened.